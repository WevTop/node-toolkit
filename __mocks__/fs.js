const fs = jest.genMockFromModule('fs')
const realFs = jest.requireActual('fs')

const fileHolder = new Map()

// Allow reading from disk.
fs.readdir = realFs.readdir
fs.readFile = realFs.readFile

// Write file contents to memory.
fs.writeFile = (path, contents, cb) => {
  fileHolder.set(path, contents)
  // In mock world, we can never fail. :')
  cb(null)
}

// Add some helper methods for getting and setting memory FS.
fs.__reset = () => fileHolder.clear()
fs.__getContents = path => fileHolder.get(path)

module.exports = fs
