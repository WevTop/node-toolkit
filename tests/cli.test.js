jest.mock('fs')

beforeEach(() => {
  require('fs').__reset()
})

test.todo('prints help when no arguments are passed')
test.todo('prints help when --help is passed')
test.todo('fails to start creating project in a directory that already exists')
test.todo('creates project with labels passed to Dockerfile from questionnaire')
