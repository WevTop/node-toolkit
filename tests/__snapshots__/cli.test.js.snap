// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates project with labels passed to Dockerfile from questionnaire: Dockerfile 1`] = `
# Use the latest version of Node.js
#
# You may prefer the full image:
# FROM node
#
# or even an alpine image (a smaller, faster, less-feature-complete image):
# FROM node:alpine
#
# You can specify a version:
# FROM node:10-slim
FROM node:slim

# Labels for GitHub to read your action
LABEL "com.github.actions.name"="My Project Name"
LABEL "com.github.actions.description"="A cool project"
# Here all of the available icons: https://feathericons.com/
LABEL "com.github.actions.icon"="anchor"
# And all of the available colors: https://developer.github.com/actions/creating-github-actions/creating-a-docker-container/#label
LABEL "com.github.actions.color"="blue"

# Copy the package.json and package-lock.json
COPY package*.json ./

# Install dependencies
RUN npm ci

# Copy the rest of your action's code
COPY . .

# Run \`node /entrypoint.js\`
ENTRYPOINT ["node", "/entrypoint.js"]

`;

exports[`creates project with labels passed to Dockerfile from questionnaire: entrypoint.js 1`] = `
const { Toolkit } = require('actions-toolkit')
const tools = new Toolkit()

console.log(tools.arguments)

`;

exports[`creates project with labels passed to Dockerfile from questionnaire: package.json 1`] = `
{
  "name": "my-project-name",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "actions-toolkit": "^1.0.0-static-version-for-test"
  }
}
`;
